eth-node/bridge/geth/envelope.go

```diff
@@ -2,29 +2,112 @@ package gethbridge
 
 import (
 	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/waku"
 	"github.com/status-im/status-go/whisper/v6"
 )
 
 type gethEnvelopeWrapper struct {
-	envelope *whisper.Envelope
+	shhEnvelope  *whisper.Envelope
+	wakuEnvelope *waku.Envelope
 }
 
-// NewGethEnvelopeWrapper returns an object that wraps Geth's Envelope in a types interface
-func NewGethEnvelopeWrapper(e *whisper.Envelope) types.Envelope {
+// NewWhisperEnvelopeWrapper returns an object that wraps Geth's Whisper Envelope in a types interface.
+func NewWhisperEnvelopeWrapper(e *whisper.Envelope) types.Envelope {
```

Comment:
This is only used once in `services/shhext/api.go`. Why not use `whisper.Enveloper` there directly?

Reply:
Because `whisper.Envelope` does not implement `Envelope` interface. An adapter is needed, that's why we have `gethEnvelopeWrapper`. What I should do is to split Whisper and Waku wrappers.

Reply:
Yes, that would make sense.